src/DoctrineBatchUtils/BatchProcessing/SimpleBatchIteratorAggregate.php

```diff
@@ -49,7 +66,7 @@ public static function fromTraversableResult(
     }
 
     /**
-     * {@inheritDoc}
+     * @return Traversable<A, B|mixed>
```

Comment:
Why `|mixed`? That defeats the whole typing 🤔

Reply:
i think i made this because of this error.

``` 
INFO: InvalidReturnType - src/DoctrineBatchUtils/BatchProcessing/SimpleBatchIteratorAggregate.php:71:16 - The declared return type 'Traversable<TKey:DoctrineBatchUtils\BatchProcessing\SimpleBatchIteratorAggregate as mixed, TValue:DoctrineBatchUtils\BatchProcessing\SimpleBatchIteratorAggregate as mixed>' for DoctrineBatchUtils\BatchProcessing\SimpleBatchIteratorAggregate::getIterator is incorrect, got 'Generator<TKey:DoctrineBatchUtils\BatchProcessing\SimpleBatchIteratorAggregate as mixed, (TValue:DoctrineBatchUtils\BatchProcessing\SimpleBatchIteratorAggregate as mixed)|object, mixed, void>' (see https://psalm.dev/011)
     * @return Traversable<TKey, TValue>
``` 

and psalm propose to fix as 

```diff
@@ -68,9 +68,11 @@
     }
 
     /**
-     * @return Traversable<TKey, TValue>
+     * @return \Generator
+     *
+     * @psalm-return \Generator<TKey, TValue|object, mixed, void>
      */
-    public function getIterator(): iterable
+    public function getIterator(): \Generator
     {
         $iteration = 0;
         $resultSet = $this->resultSet;

``` 

I think the part `TValue|object, mixed, void` is because of  how `getIterator` is design (re fetch object, if its array take the first elements etc...).
I am not a an pslam expert so have you any idea ?

Reply:
Ping!